### PR TITLE
[Backport 2.x] Improve string terms aggregation performance using Collector#setWeight

### PR DESCRIPTION
**Pull request by sandeshkr419**
**TLDR:** The PR is a combination of 2 PRs merged in `main`: https://github.com/opensearch-project/OpenSearch/pull/11643 & https://github.com/opensearch-project/OpenSearch/pull/12643
### Description
Test automatically merges the index after delete sometimes leading to flaky test. Avoiding that from happening to stabilize test.
Follow-up on https://github.com/opensearch-project/OpenSearch/pull/12628 
Backport https://github.com/opensearch-project/OpenSearch/commit/7dac98c81ae387e5d4782a9599505a4c376a3e74 from https://github.com/opensearch-project/OpenSearch/pull/11643.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by sandeshkr419** [comment]
@msfroh Closed https://github.com/opensearch-project/OpenSearch/pull/12628 since 2.x was somehow merging up test indices. Modified test-index a bit to not merge at all - see the additional commit.
I'm unsure about index policy changes between 2.x and 3.0/main since the test is not at all flaky in main, so not adding the cherry-commit of this PR to `main` branch at the moment.

**Review by deshsidd — APPROVED** [review]

**Comment by sandeshkr419** [comment]
@reta Can you check (& possibly merge) this - I see you already have context here.

**Comment by reta** [comment]
> @reta Can you please check (& possibly merge) this - I see you already have context here.

I haven't checked the original pull request closely, I would leave it to @msfroh

**Review by msfroh — APPROVED** [review]
